components/scheduler2/core/scheduler.go

```diff
@@ -0,0 +1,171 @@
+package core
```

Comment:
I wouldn't create this extra package, just put everything in `components/scheduler`, it'll help you avoid circular dependencies later and allow you to not expose anything that's not strictly necessary.

Reply:
The reason it's separated out is because the substate machines would have a lot of conflicts both between boilerplate (`generator.MachineKey`/`backfiller.MachineKey` -> `GeneratorMachineKey`/`BackfillerMachineKey`) as well as conflict between states/events (`generator.TransitionSleep`/`executor.TransitionSleep` -> `TransitionGeneratorSleep`/`TransitionExecutorSleep`). It also helps avoid creating circular dependencies and tight coupling between components. 

I can be convinced the other way if it's going to be the standard for HSM projects, to keep all within a single package, but if there *is* a component that were to be fitting for multiple component packages, I think it'd be this one.

Reply:
Hmm... will leave it up to you.

Reply:
As a result of a bunch of the event/state compaction, I think the conflicts between components are much more minimal, so I'm going to end up moving it all into the one package.